src/uu/ls/src/ls.rs

```diff
@@ -537,19 +539,45 @@ fn extract_time(options: &clap::ArgMatches) -> Time {
     }
 }
 
+/// Parse the dircolors terminal list into a RegexSet and check against
+/// current terminal if applicable
+///
+/// # Returns
+///
+/// Boolean
+fn check_env_variables_against_dircolors(dircolors: &str) -> bool {
```

Comment:
I think this name could be more descriptive about what it's checking. If I understand correctly, it's checking that the terminal supports colors?

Reply:
I also think the `lscolors` crate might be interested in a feature like this. It looks like something they might want to support.

Reply:
@tertsdiepraam LSCOLORS would need to hardcode the values since DIRCOLORS as in this repo does not exist. I can open an issue ofc.
Back to topic, we are not actually checking against `is_terminal` which is already provided in many places, `LS` is using the terminal list hardcoded in `DIRCOLORS.h` to check if the terminal provided via `TERM` environment variable is in that list. Sadly even the `--color=always` does this, so it basically works almost like `--color=auto`.
In addition to this check `LS` and rust `LS` are checking for terminal already.

I will think of a more descriptive name, I struggled because the whole topic is verymuch down the rabbit hole.